pkg/trace/api/api.go

```diff
@@ -596,6 +603,17 @@ func (r *HTTPReceiver) handleTraces(v Version, w http.ResponseWriter, req *http.
 	}
 }
 
+const outOfCpuLogThreshold uint32 = 5
```

Comment:
I'm not super confident about this threshold, should we make it configurable?

Reply:
Do we have an idea on how many times this log will happen when the agent is under heavy load? Just seeing the log will be really useful for a customer so as long as we aren't accidentally helping to overwhelm the agent here by over-logging this warning (which I don't think we are?)

Reply:
Yea I'm mostly worried about it becoming spammy, at the same time we're spawning goroutines withouth limits here so we really have to let the user/support know there is an issue 😅 

> Do we have an idea on how many times this log will happen when the agent is under heavy load?
I lowered the payload channel to 50 instead off 1000 to speed-up the reproducer - I saw the warning logged 3 times in a row
```
2022-08-29 18:44:47 UTC | TRACE | WARN | (run.go:261 in Warnf) | The Agent is falling behind on processing traces, 1 extra threads have been created since the Agent started. See https://docs.datadoghq.com/tracing/troubleshooting/agent_apm_resource_usage
2022-08-29 18:44:47 UTC | TRACE | WARN | (run.go:261 in Warnf) | The Agent is falling behind on processing traces, 6 extra threads have been created since the Agent started. See https://docs.datadoghq.com/tracing/troubleshooting/agent_apm_resource_usage
2022-08-29 18:44:47 UTC | TRACE | WARN | (run.go:261 in Warnf) | The Agent is falling behind on processing traces, 11 extra threads have been created since the Agent started. See https://docs.datadoghq.com/tracing/troubleshooting/agent_apm_resource_usage
```
And then the Agent started to catch-up.
I think we'd see more warning logs in a real-life scenario (i.e with 1000 buffered payloads) so maybe we should increase `outOfCpuLogThreshold` a bit.